test/smoke/testApps/CustomInstrumentation/src/smokeTest/java/com/microsoft/applicationinsights/smoketestapp/CustomInstrumentationTest.java

```diff
@@ -47,12 +44,9 @@ public void customInstrumentationOne() {
 
     @Test
     @TargetUri("/customInstrumentationTwo")
-    public void customInstrumentationTwo() {
-        List<Envelope> rdList = mockedIngestion.getItemsEnvelopeDataType("RequestData");
-        List<Envelope> rddList = mockedIngestion.getItemsEnvelopeDataType("RemoteDependencyData");
-
-        assertThat(rdList, hasSize(1));
-        assertThat(rddList, hasSize(1));
+    public void customInstrumentationTwo() throws Exception {
+        List<Envelope> rdList = mockedIngestion.waitForItems("RequestData", 1);
+        List<Envelope> rddList = mockedIngestion.waitForItems("RemoteDependencyData", 1);
```

Comment:
Before this change, if !=1 items were received it would throw AssertionException causing a test failure. Now this will throw IllegalStateException and test fails with an error. Perhaps waitForItems should fail with some type of AssertionException instead. The tearDown flow seems to be more stable when it follows that path.

Perhaps in the future we could have something like: `mockedIngestion.assertReceived(numberOfItems(1), ofType("RequestData"))`, but we don't have to do this now.

Reply:
👍 I made changed from `IllegalStateException` to `AssertionError`, and improved message in the standard form of "expected X but received Y".